src/icon-manifest/iconGenerator.ts

```diff
@@ -375,9 +375,30 @@ export class IconGenerator implements IIconGenerator {
 
   private getIconPath(defaultPath: string, filename: string): string {
     const absPath = path.join(this.settings.vscodeAppData, this.settings.extensionSettings.customIconFolderName);
-    return !this.avoidCustomDetection && this.hasCustomIcon(absPath, filename)
-      ? this.getRelativePath(this.manifestFolderPath, absPath, false)
-      : defaultPath;
+    if (!this.avoidCustomDetection && this.hasCustomIcon(absPath, filename)) {
+      const sanitizedFolderPath =
+        this.belongToSameDrive(absPath, this.manifestFolderPath) ?
+          this.manifestFolderPath :
+          this.overwriteDrive(absPath, this.manifestFolderPath);
+      return this.getRelativePath(sanitizedFolderPath, absPath, false);
+    } else {
```

Comment:
When you use a `return` statement inside an `if` condition, the use of `else` is redundant.